cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -129,6 +129,12 @@ public E from(S source) {
         return (E) this;
     }
 
+    public E withId(int id) {
+        this.id = id;
+        //noinspection unchecked
```

Comment:
There is a common `returnThis()` pattern that can avoid this unchecked cast. I don't this use warrants a change, just mentioning it for future reference.